src/uvw/resource.hpp

```diff
@@ -29,19 +29,19 @@ class Resource: public Emitter<T>, public std::enable_shared_from_this<T> {
           pLoop{std::move(ref)},
           resource{}
     {
-        this->template get<U>()->data = static_cast<T*>(this);
+        resource.data = static_cast<T*>(this);
     }
 
     auto parent() const noexcept {
         return pLoop->loop.get();
     }
 
-    template<typename R>
+    template<typename R = U>
```

Comment:
I'll make the suggested changes.
I have a quick question though: do we need the `static_cast` in the `const` qualified non-template `get`? I think we could make do with just a `&resource` expression.
Ideas?